nuxeo-multi-tenant-core/src/test/java/org/nuxeo/ecm/multi/tenant/TestMultiTenantDirectories.java

```diff
@@ -0,0 +1,161 @@
+@Deploy("org.nuxeo.ecm.multi.tenant")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-enabled-default-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-directories-contrib.xml")
+@Deploy("org.nuxeo.ecm.platform.test:test-usermanagerimpl/userservice-config.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD, init = MultiTenantRepositoryInit.class)
+public class TestMultiTenantDirectories {
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected DirectoryService directoryService;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void iCanCreateSameDirectoryEntryInDifferentTenant() throws LoginException {
+        // Add user0 and user1 to the powerusers group so they can edit the directory
+        Framework.doPrivileged(() -> {
+            NuxeoPrincipal user0Principal = userManager.getPrincipal("user0");
+            List<String> groups = user0Principal.getGroups();
+            groups.add(Constants.POWER_USERS_GROUP);
+            user0Principal.setGroups(groups);
+            userManager.updateUser(user0Principal.getModel());
+            NuxeoPrincipal user1Principal = userManager.getPrincipal("user1");
+            groups = user1Principal.getGroups();
+            groups.add(Constants.POWER_USERS_GROUP);
+            user1Principal.setGroups(groups);
+            userManager.updateUser(user1Principal.getModel());
+        });
+
+        // Map of the entry to create per tenant
+        Map<String, Object> newEntry = new HashMap<>();
+        newEntry.put("id", "key0");
+        newEntry.put("label", "Label Key 0");
+
+        // As user0, add a new entry in directory
+        LoginContext loginContext = Framework.loginAsUser("user0");
+        try (CloseableCoreSession userSession = openSession()) {
+            // Open the testDirectory
+            try (Session sessionDir = directoryService.open("testDirectory")) {
+                sessionDir.createEntry(newEntry);
+            }
+        }
+        loginContext.logout();
+
+        // As user1, add the same entry in directory
+        loginContext = Framework.loginAsUser("user1");
+        try (CloseableCoreSession userSession = openSession()) {
+            // Open the testDirectory
+            try (Session sessionDir = directoryService.open("testDirectory")) {
+                sessionDir.createEntry(newEntry);
+            }
+        }
+        loginContext.logout();
+
+        // Check that the two entries are created
+        Framework.doPrivileged(() -> {
+            // Open the testDirectory
+            try (Session sessionDir = directoryService.open("testDirectory")) {
+                DocumentModelList entries = sessionDir.query(Collections.emptyMap());
+                assertThat(entries).hasSize(2);
+            }
+        });
+    }
+
+    @Test(expected = DirectoryException.class)
```

Comment:
Could you not rely on `expected` and do a try/catch with an assert on message ? (if possible)

Reply:
This will also enforce fact that test is failing at second creation.

Reply:
Done